src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java

```diff
@@ -1,129 +1,131 @@
-	}
-
-	private static final Map<TestExecutionResult.Status, String> STATUS_MAPPING = ImmutableMap.<TestExecutionResult.Status, String>builder()
-			.put(TestExecutionResult.Status.SUCCESSFUL, Statuses.PASSED)
-			.put(TestExecutionResult.Status.ABORTED, Statuses.FAILED)
-			.put(TestExecutionResult.Status.FAILED, Statuses.FAILED)
-			.build();
-
-	//	testIdentifier.getSource().ifPresent(source -> {
-	//		if (source instanceof MethodSource) {
-	//			MethodSource methodSource = (MethodSource) source;
-	//			Class<?> testClass = Class.forName(methodSource.getClassName());
-	//			Method testMethod = testClass.getDeclaredMethod(methodSource.getMethodName());
-	//			Optional<RepeatedTest> annotation = AnnotationUtils.findAnnotation(testMethod, RepeatedTest.class);
-	//
-	//		}
-	//	});
-}
+/*
+ * Copyright 2018 EPAM Systems
+ *
+ *
+ * This file is part of EPAM Report Portal.
+ * https://github.com/reportportal/commons-model
+ *
+ * Report Portal is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Report Portal is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Report Portal.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.epam.reportportal.junit5;
+
+import com.epam.reportportal.listeners.ListenerParameters;
+import com.epam.reportportal.listeners.Statuses;
+import com.epam.reportportal.service.Launch;
+import com.epam.reportportal.service.ReportPortal;
+import com.epam.ta.reportportal.ws.model.FinishExecutionRQ;
+import com.epam.ta.reportportal.ws.model.FinishTestItemRQ;
+import com.epam.ta.reportportal.ws.model.StartTestItemRQ;
+import com.epam.ta.reportportal.ws.model.launch.StartLaunchRQ;
+import io.reactivex.Maybe;
+
+import org.junit.jupiter.api.extension.Extension;
+import org.junit.platform.engine.TestExecutionResult;
+import org.junit.platform.engine.TestTag;
+import org.junit.platform.launcher.TestExecutionListener;
+import org.junit.platform.launcher.TestIdentifier;
+import org.junit.platform.launcher.TestPlan;
+import rp.com.google.common.collect.ImmutableMap;
+
+import java.util.Calendar;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.stream.Collectors;
+
+import static java.util.Optional.ofNullable;
+
+/**
+ * ReportPortal Listener sends the results of test execution to ReportPortal in RealTime
+ *
+ * @author <a href="mailto:andrei_varabyeu@epam.com">Andrei Varabyeu</a>
+ */
+public class ReportPortalExtension implements TestExecutionListener, Extension {

```

Comment:
This revision is based on the ServiceLoader configuration of this project. If the `Extension` marker interface isn't truly applicable, this revision should be reverted and the corresponding ServiceLoader configuration files should be removed.